src/index.d.ts

```diff
@@ -525,6 +525,23 @@ declare namespace RamdaAdjunct {
         mapIndexed(iterator: (char: string, key: number, str: string) => string, str: string): string[];
         mapIndexed(iterator: (char: string, key: number, str: string) => string): (str: string) => string[];
 
+        /**
+         * {@link http://ramdajs.com/docs/#reduce|R.reduce} function that more closely resembles Array.prototype.reduce.
+         * It takes two new parameters to its callback function: the current index, and the entire list.
+         */
+        reduceIndexed<T, TResult, R extends T[]>(
+            iterator: (acc: TResult, elem: T, key: number, list: R) => TResult,
+            acc: TResult,
+            list: R,
+        ): TResult;
+        reduceIndexed<T, TResult, R extends T[]>(
+            iterator: (acc: TResult, elem: T, key: number, list: R) => TResult,
+            acc: TResult,
+        ): (list: R) => TResult;
+        reduceIndexed<T, TResult, R extends T[]>(
+            iterator: (acc: TResult, elem: T, key: number, list: R) => TResult,
+        ): (acc: TResult) => (list: R) => TResult;
```

Comment:
https://www.typescriptlang.org/play/index.html#src=declare%20function%20reduceIndexed%3CT%2C%20TResult%2C%20R%20extends%20T%5B%5D%3E(%0D%0A%20%20%20%20iterator%3A%20(acc%3A%20TResult%2C%20elem%3A%20T%2C%20key%3A%20number%2C%20list%3A%20R)%20%3D%3E%20TResult%2C%0D%0A%20%20%20%20acc%3A%20TResult%2C%0D%0A%20%20%20%20list%3A%20R%2C%0D%0A)%3A%20TResult%3B%0D%0Adeclare%20function%20reduceIndexed%3CT%2C%20TResult%2C%20R%20extends%20T%5B%5D%3E(%0D%0A%20%20%20%20iterator%3A%20(acc%3A%20TResult%2C%20elem%3A%20T%2C%20key%3A%20number%2C%20list%3A%20R)%20%3D%3E%20TResult%2C%0D%0A%20%20%20%20acc%3A%20TResult%2C%0D%0A)%3A%20(list%3A%20R)%20%3D%3E%20TResult%3B%0D%0Adeclare%20function%20reduceIndexed%3CT%2C%20TResult%2C%20R%20extends%20T%5B%5D%3E(%0D%0A%20%20%20%20iterator%3A%20(acc%3A%20TResult%2C%20elem%3A%20T%2C%20key%3A%20number%2C%20list%3A%20R)%20%3D%3E%20TResult%2C%0D%0A)%3A%20%7B%0D%0A%20%20%20%20%20%20%20%20(acc%3A%20TResult)%3A%20(list%3A%20R)%20%3D%3E%20TResult%3B%0D%0A%20%20%20%20%20%20%20%20(acc%3A%20TResult%2C%20list%3A%20R)%3A%20TResult%0D%0A%7D%3B%0D%0A%0D%0Alet%20x1%20%3D%20reduceIndexed((acc%3A%20number%2C%20elem%3A%20number)%20%3D%3E%20acc%20%2B%20elem%2C%200%2C%20%5B1%2C%202%2C%203%5D)%0D%0Alet%20x2%20%3D%20reduceIndexed((acc%3A%20number%2C%20elem%3A%20number)%20%3D%3E%20acc%20%2B%20elem%2C%200)(%5B1%2C%202%2C%203%5D)%0D%0Alet%20x3%20%3D%20reduceIndexed((acc%3A%20number%2C%20elem%3A%20number)%20%3D%3E%20acc%20%2B%20elem)(0%2C%20%5B1%2C%202%2C%203%5D)%0D%0Alet%20x4%20%3D%20reduceIndexed((acc%3A%20number%2C%20elem%3A%20number)%20%3D%3E%20acc%20%2B%20elem)(0)(%5B1%2C2%2C3%5D)%0D%0A

you should add one case

Good work